admin-tool/src/aio/configure.rs

```diff
@@ -46,6 +46,11 @@ pub(super) struct Configuration {
     #[clap(long)]
     pub manufacturing_use_secp256r1: bool,
 
+    #[clap(long)]
```

Comment:
Just a question: Is this how the clap cli parser is implemented? I've read a little about it a while ago but haven't seen it used before.

Reply:
yes, if you look at the top of the struct there are a couple of `Derive`s:
```rust
#[derive(Debug, Args, Clone, Serialize, Deserialize)]
```
 `Debug` and `Clone` are Rust-specific derives, `Serialize` and `Deserialize` are from Serde (see the imports) and `Args` comes from Clap (see the import).
With Clap you can use Derives to define arguments to the cli functions that you will be implementing if you define in the Cargo.toml that you want the 'Derive' feature (see: https://github.com/fdo-rs/fido-device-onboard-rs/blob/main/admin-tool/Cargo.toml#L15). Then, with the `Args` derive you are telling Clap that those are arguments to cli function, and with the specific `#[clap(long)]`, `#[clap(whaetever)]` you're telling Clap how to parse the options.
There is a small example on how to add a new cli function here: https://github.com/fdo-rs/fido-device-onboard-rs/commit/4da0a49bf9d7f272b86b07e009b0dde3bfc5600d
And a more complete example on how you would implement a full cli with Clap from scratch (`manufacturing-client/src/main.rs` is the relevant file): https://github.com/fdo-rs/fido-device-onboard-rs/commit/dfe51b2803d2a8aa47ef939130c7391426def1e6